make start creates root-owned .github/test/ssl/ when run before generate-certs

## Problem

Running `make start` (or any compose-up path) in `.github/test/` **before** `make generate-certs` produces an empty `.github/test/ssl/` directory owned by **root**, blocking subsequent cert generation as a non-root user.

The next agent that comes up then fails because the read-only `ssl/` mount is empty:

```
Caused by: java.io.FileNotFoundException: ssl/server.crt (No such file or directory)
```

…and `make generate-certs` cannot write into the root-owned dir, so the user must `sudo rm -rf .github/test/ssl` to recover.

## Root cause

`ssl/` is gitignored (`./.github/test/.gitignore` line 2). Three agent compose files mount it read-only:

- `.github/test/tc-agent/compose.yaml:12` — `../ssl:/app/ssl:ro`
- `.github/test/cd-agent/compose.yaml:14` — `../ssl:/app/ssl:ro`
- `.github/test/rd-agent/compose.yaml:12` — `../ssl:/app/ssl:ro`

When the bind-mount source is missing, the Docker daemon (running as root) auto-creates the directory — owned by root. `make all` works because the target chain is `pull generate-certs start ...`, but `make start` alone, or any direct `docker compose up`, hits this footgun.

## Suggested fix

Add `generate-certs` (or a guard target that runs it only when certs are missing) as a prerequisite of `start` in `.github/test/Makefile`. Either:

```makefile
start: generate-certs
	docker compose ${COMPOSE_FILES} up --wait --pull missing ${SERVICES}
```

…or a lazy guard:

```makefile
ssl/server.crt:
	$(MAKE) generate-certs

start: ssl/server.crt
	docker compose ${COMPOSE_FILES} up --wait --pull missing ${SERVICES}
```

The latter avoids regenerating certs on every `make start`.

## Workaround

```sh
sudo rm -rf .github/test/ssl
make -C .github/test generate-certs start
```

## Found while

Wiring local-data support into `measure-e2e.sh`. The script's `ensure_stack_up` previously called `make start` directly; it now runs `make generate-certs` when `ssl/server.crt` is missing, but the underlying Makefile trap remains.

--- a/.github/test/Makefile
+++ b/.github/test/Makefile
@@ -21,7 +21,10 @@ generate-certs:
 	  | openssl x509 -req -CA ca.crt -CAkey ca.key -CAcreateserial -out server-rd-agent.crt -days 30 && \
 	  chmod +r server-rd-agent.key)
 
-start:
+ssl/server.crt:
+	$(MAKE) generate-certs
+
+start: ssl/server.crt
 	docker compose ${COMPOSE_FILES} up --wait --pull missing ${SERVICES}
 
 download-test-data-checksums:

